tools/kagami/src/crypto.rs

```diff
@@ -1,14 +1,201 @@
-use clap::{builder::PossibleValue, ArgGroup, ValueEnum};
-use color_eyre::eyre::WrapErr as _;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use clap::{builder::PossibleValue, ArgGroup, Subcommand, ValueEnum};
+use color_eyre::{
+    eyre::{eyre, WrapErr as _},
+    Result,
+};
 use iroha_crypto::{Algorithm, KeyPair, PrivateKey};
+use iroha_data_model::ChainId;
+use iroha_genesis::{GenesisNetwork, RawGenesisBlock};
+use parity_scale_codec::Encode;
 
 use super::*;
 
+const ED25519_PREFIX: &str = "ed0120";
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum Args {
+    SignGenesis(SignGenesisArgs),
+    GenerateKeyPair(GenerateKeyPairArgs),
+}
+
+/// Use `Kagami` to sign genesis block.
+#[derive(ClapArgs, Clone, Debug)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+#[command(group = ArgGroup::new("format").required(false))]
+pub struct SignGenesisArgs {
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: AlgorithmArg,
+    /// Private key (in string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
+    /// Path to private key to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_file: Option<PathBuf>,
+    /// Public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_string: Option<String>,
+    /// Path to public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_file: Option<PathBuf>,
+    /// Path to json-serialized keypair
+    #[clap(long, short, group = "private_key", group = "public_key")]
+    keypair_file: Option<PathBuf>,
+    /// Unique id of blockchain
+    #[clap(long)]
+    chain_id: ChainId,
+    /// Path to genesis json file
+    #[clap(long, short)]
+    genesis_file: PathBuf,
+    /// Output signed genesis block in JSON format
+    #[clap(long, short, default_value_t = true, group = "format")]
+    json: bool,
+    /// Encode signed genesis block with SCALE (it is only supported with file output)
+    #[clap(long, short, default_value_t = false, group = "format")]
+    scale: bool,
+    /// Path to signed genesis output file (stdout by default)
+    #[clap(long, short)]
+    outfile: Option<PathBuf>,
```

Comment:
```suggestion
    out_file: Option<PathBuf>,
```

@0x009922 we seem to be mixing these 2. I prefer `out_file` but it's fine either way as long as it's consistent

Reply:
Agree. I prefer `--out-file` too.